### PR TITLE
Use image server for all kind of images and integrate identity rings

### LINKED ISSUE #13670: Use image server for all kind of images and integrate identity rings
### Description

Currently the http server in status-go https://github.com/status-im/status-go/blob/develop/server/server.go is used only for identicon and image messages.
We should also use it for profile pictures. 

There's a branch out that also handles this case, which includes work from @qfrank:
https://github.com/status-im/status-mobile/tree/feature/identity-images-http-server
https://github.com/status-im/status-go/tree/feature/13334

This branch has a few changes:

- Adds an identity ring to the image
- Adds a signal when the port of the http server changes
- Serves user profile pictures

Some of the work integrating the branch has already been done here:

The important things to note when finishing integrating the branch:

- If the port of the server changes, the URL should of the image should be updated and re-fetched
- Status-go should not serve user images if the user has their settings set to not display profile pictures, only identicons should be server
- If this setting changes, pictures should be reloaded
- Dark/light theme should be likely passed as a parameter, so that the identity ring can be correctly set
- Identity ring should be toggled, as we don't want to show it in the old UI, only on the new UI

### PR DESCRIPTION
**Pull request by qfrank**
Closes [#13670](https://github.com/status-im/status-mobile/issues/13670)

### CODE REVIEW

**Comment by qoqobolo** [comment]
Hello @qfrank, thanks for your work on this!
Is this ready to be reviewed? If so, please move PR to the `Review` column on the board and send requests to reviewers. 
We'll start testing the related status-mobile PR https://github.com/status-im/status-mobile/pull/13771 after getting approval on the go-side and moving it to the `Ready for client testing` column.
Thank you!

**Comment by qfrank** [comment]
> 

hi @qoqobolo , it's ready for review :)

File: images/decode.go
Comment (Samyoul) [inline images/decode.go]: Would you be able to introduce a test for this function? 🙂

File: mobile/status.go
Comment (Samyoul) [inline mobile/status.go]: The import convention (and zealously enforced by the linter 😆) in this repo is to group imports alphabetically as follows:

- standard lib
- 3rd party libs
- core libs
   - ethereum/go-ethereum
   - status-im/status-go

File: server/handlers.go
Comment (Samyoul) [inline server/handlers.go]: Same as https://github.com/status-im/status-go/pull/2788/files#r943285171

File: server/handlers.go
Comment (Samyoul) [inline server/handlers.go]: `enableRing` reads like a command but returns a bool, perhaps `ringEnabled` communicates the question this function answers.

File: server/handlers.go
Comment (Samyoul) [inline server/handlers.go]: This could potentially be simplified to:

```suggestion
	return ok && len(addRings) == 1 && addRings[0] == "1"
```

File: mobile/status.go
Comment (qfrank) [reply mobile/status.go]: emm.. i thought `make lint-fix` would fix this, i was wrong, thanks for pointing out!

File: images/decode.go
Comment (qfrank) [reply images/decode.go]: this function can be removed now, we don't need it anymore, since we don't want to transfer images through the bridge anymore

**Comment by qfrank** [comment]
Hi @Samyoul , ready to review again :)

File: server/server.go
Comment (Samyoul) [inline server/server.go]: This is triggered inside the `Server.listenAndServe` function, this will trigger for all types of `Server` including the `PairingServer`

Comment (qfrank) [reply server/server.go]: Hi @Samyoul , thanks for pointing out, just made a small fix.

**Review by Samyoul — APPROVED** [review]
Nice PR, thank you for addressing my points so fast

**Pull request merged**
